File: PHP/hello_world_again/src/Golem.php
<?php


class Golem
{
    public int $hp = 0;
    public int $defense = 0;
    public int $attack = 0;

    public function __construct(int $hp = 300, int $defense = 80, int $attack = 50)
    {
        $this->hp = $hp;
        $this->defense = $defense;
        $this->attack = $attack;
    }
}

File: Elixir/hello_world_again/lib/shared/golem.ex
defmodule Golem do
  defstruct [
    :hp,
    :defense,
    :attack
  ]

  @type hp :: integer()
  @type defense :: integer()
  @type attack :: integer()

  @type t :: %__MODULE__{
    hp: hp,
    defense: defense,
    attack: attack
  }

  @spec new(integer(), integer(), integer()) :: Golem.t()
  def new(hp \\ 300, defense \\ 80, attack \\ 50) do
    %__MODULE__{
      hp: hp,
      defense: defense,
      attack: attack
    }
  end
end
